lib/include/openturns/OTtestcode.hxx

```diff
@@ -454,6 +429,596 @@ void assert_equal(const T & a, const T & b, const String errMsg = "")
+    if (enableGradient_)
+    {
+      checkPDFGrad();
+      checkCDFGrad();
+    }
+    checkMinimumVolumeInterval();
+    if (enableMinimumVolumeLevelSet_)
+      checkMinimumVolumeLevelSet();
+    checkConfidenceInterval();
+    if (enableCharacteristicFunction_)
+      checkCharacteristicFunction();
+  }
+
+  void skipCDF() { enableCDF_ = false; }
+  void skipDDF() { enableDDF_ = false; }
+  void skipMoments() { enableMoments_ = false; }
+  void skipCorrelation() { enableCorrelation_ = false; }
+  void skipMinimumVolumeLevelSet() { enableMinimumVolumeLevelSet_ = false; }
+  void skipParameters() { enableParameters_ = false; }
+  void skipProbability() { enableProbability_ = false; }
+  void skipCharacteristicFunction() { enableCharacteristicFunction_ = false; }
+  void skipGradient() { enableGradient_ = false; }
+  void skipEntropy() { enableEntropy_ = false; }
+  void setEntropyTolerance(const Scalar entropyTolerance) { entropyTolerance_ = entropyTolerance; }
+  void setEntropySamplingSize(const UnsignedInteger entropySamplingSize) { entropySamplingSize_ = entropySamplingSize; }
+  void setMeanTolerance (const Scalar meanTolerance) { meanTolerance_ = meanTolerance; }
+  void setStandardDeviationTolerance (const Scalar standardDeviationTolerance) { standardDeviationTolerance_ = standardDeviationTolerance; }
+  void setSkewnessTolerance (const Scalar skewnessTolerance) { skewnessTolerance_ = skewnessTolerance; }
+  void setKurtosisTolerance (const Scalar kurtosisTolerance) { kurtosisTolerance_ = kurtosisTolerance; }
+  void setCorrelationTolerance (const Scalar correlationTolerance) { correlationTolerance_ = correlationTolerance; }
+  void setMomentsSamplingSize(const Scalar momentsSamplingSize) { momentsSamplingSize_ = momentsSamplingSize; }
+
+private:
+
+  void checkPrint() const
+  {
+    LOGTRACE(OSS() << distribution_.__repr__());
+    LOGTRACE(OSS() << distribution_.__str__());
+    LOGTRACE(OSS() << distribution_.__repr_markdown__());
+    LOGTRACE(OSS() << distribution_._repr_html_());
+  }
+  
+  void checkGeneral() const
+  {
+    LOGTRACE(OSS() << "elliptical=" << distribution_.isElliptical());
+    LOGTRACE(OSS() << "continuous=" << distribution_.isContinuous());
+    LOGTRACE(OSS() << "discrete=" << distribution_.isDiscrete());
+    LOGTRACE(OSS() << "integral=" << distribution_.isIntegral());
+    LOGTRACE(OSS() << "copula=" << distribution_.isCopula());
+
+    Distribution standardRep(distribution_.getStandardRepresentative());
+    LOGTRACE(OSS() << "Standard representative=" << standardRep);
+
+    if (distribution_.isDiscrete() && (distribution_.getDimension() == 1))
+    {
+      const Complex z(0.3, 0.7);
+      Complex GF = distribution_.computeGeneratingFunction(z);
+      LOGTRACE(OSS() << "generating function=" << GF);
+      Complex LGF = distribution_.computeLogGeneratingFunction(z);
+      LOGTRACE(OSS() << "log generating function=" << LGF);
+    }
+  }
+
+  void checkCharacteristicFunction() const
```

Comment:
This does not require much to the characteristic function (see [Wikipedia](https://en.wikipedia.org/wiki/Characteristic_function_(probability_theory))).
- We may check that $\varphi(0) = 1$.
- We may check that $\mathbb{E}[X^k] = i^{-k} \varphi^{(k)}(0)$. This may require to compute the derivative function of the characteristic function, e.g. using finite differences.

Reply:
maybe the first item then